rpcs3/Emu/NP/np_requests.cpp

```diff
@@ -1086,6 +1088,23 @@ namespace np
 		transaction_async_handler(std::move(lock), trans_ctx, req_id, async);
 	}
 
+	template <typename T>
+	void set_rankdata_values(T& cur_rank, const ScoreRankData* fb_rankdata)
```

Comment:
Consider turning it into `vm::ptr<T>& cur_rank` and then the `is_same_v` check below can be simpler.